dali/operators/decoder/numpy.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_DECODER_NUMPY_H_
+#define DALI_OPERATORS_DECODER_NUMPY_H_
+
+#include <vector>
+
+#include "dali/core/common.h"
+#include "dali/core/error_handling.h"
+#include "dali/pipeline/operator/checkpointing/stateless_operator.h"
+#include "dali/pipeline/operator/operator.h"
+#include "dali/util/numpy.h"
+
+namespace dali {
+
+class NumpyDecoder : public StatelessOperator<CPUBackend> {
+ public:
+  explicit inline NumpyDecoder(const OpSpec &spec) : StatelessOperator<CPUBackend>(spec) {
+    if (spec.HasArgument("dtype")) {
+      dtype_ = spec.GetArgument<DALIDataType>("dtype");
+    } else {
+      // If dtype is not specified, it will be inferred from the input data
+      dtype_ = std::nullopt;
+    }
+  }
+
+  inline ~NumpyDecoder() override = default;
+  DISABLE_COPY_MOVE_ASSIGN(NumpyDecoder);
+
+  bool HasContiguousOutputs() const override {
+    return false;
+  }
```

Comment:
```suggestion
```

The default true should work here just fine, I think. You know the shapes and type during Setup and then 1:1 copy individual samples into the output. (The samples shapes don't have to be uniform if that was your concern).